rudder-core/src/main/scala/com/normation/rudder/services/policies/SystemVariableService.scala

```diff
@@ -356,12 +361,57 @@ class SystemVariableServiceImpl(
      * Cf3PromisesFileWriterServiceImpl#prepareRulesForAgents
      */
     val varNodeConfigVersion = systemVariableSpecService.get("RUDDER_NODE_CONFIG_ID").toVariable(Seq("DUMMY NODE CONFIG VERSION"))
+
+
+    /*
+     * RUDDER_NODE_GROUPS_VAR is an array of group_uuid -> group_name for the node
+     * RUDDER_NODE_GROUPS_CLASSE are pairs of group_UUID, group_NORMALIZED_NAME,
+     * for ex if node belongs to group:
+     * (id: 64f85ba8-39c7-418a-a099-24c2c2909dfd ; name: "Serveurs pré-prod")
+     * we will have the following classes:
+     *   - group_64f85ba8_39c7_418a_a099_24c2c2909dfd
+     *   - group_Serveur_pre_prod
+     * and var:
+     *   - by_uuid["64f85ba8-39c7-418a-a099-24c2c2909dfd"] string => "Serveurs pré-prod"
+     */
+    //build the list of nodeId -> names, taking care of special nodeIds for special target
+    val nodeGroups = allGroups.getTarget(nodeInfo.id, allNodeInfos).map { case(target, info) =>
+      val id = info.target.target match {
+        case GroupTarget(id) => id.value
+        case t => t.target
+      }
+      (id, info.name)
+    }
+    val nodeMaxString = if(nodeGroups.isEmpty) 0 else nodeGroups.flatMap { case (a,b) => a.size :: b.size :: Nil }.max
+    val stringNodeGroupsVars = if(nodeGroups.isEmpty) {
+      ""
+    } else {
+      nodeGroups.map { case (id, name) =>
+        s"""by_uuid["${id}"] ${" "*(nodeMaxString-id.size)} string => "${name}",\n""" +
+        s"""            ${" "*(nodeMaxString)        }   meta => { "inventory", "attribute_name=rudder_groups" };"""
```

Comment:
oups. And finally, I thought that the code may not evolve that much, because the name of the var will be freezed as soons as we release the version (public API). And other changes would probably need a rudder update, to (so not grave if in scala code). 
Also, it seems that the pasing of template with variables/loop is much longer (we are talking about a few ms, and in a unique file, so that may not be relevant - especially given the fiability of the test). 
And vertical alignement was much harder to do, to (not a real blocker ;)

So, all in all, it seems that the hypothetic futur gain compared present loss was not in favor of the previous choice. I can still try to implement it if you are thinking it is the best solution.